src/components/MultiStepIndicator/MultiStepIndicator.tsx

```diff
@@ -12,32 +9,58 @@ import { SIZES, MULTI_STEP_TYPES, STEP_STATUSES, TEXT_PLACEMENTS } from "./Multi
 import { getTestId } from "../../tests/test-ids-utils";
 import { ComponentDefaultTestId } from "../../tests/constants";
 import styles from "./MultiStepIndicator.module.scss";
+import { withStaticProps, VibeComponentProps, VibeComponent, SubIcon } from "../../types";
+import { IconType } from "../Icon/IconConstants";
 
-const MultiStepIndicator = forwardRef(
+export type Step = {
+  titleText: string;
+  subtitleText: string;
+  status: STEP_STATUSES;
+};
+
+interface MultiStepIndicatorProps extends VibeComponentProps {
+  steps: Step[];
+  type?: MULTI_STEP_TYPES;
+  stepComponentClassName?: string;
+  dividerComponentClassName?: string;
+  fulfilledStepIcon?: SubIcon;
+  fulfilledStepIconType?: IconType.SVG | IconType.ICON_FONT;
+  isFulfilledStepDisplayNumber?: boolean;
+  onClick?: (stepNumber: number) => void;
+  textPlacement?: TEXT_PLACEMENTS;
+  size?: SIZES;
+}
+
+const MultiStepIndicator: VibeComponent<MultiStepIndicatorProps> & {
+  types?: typeof MULTI_STEP_TYPES;
+  stepStatuses?: typeof STEP_STATUSES;
+  textPlacements?: typeof TEXT_PLACEMENTS;
+  sizes?: typeof SIZES;
+} = forwardRef(
   (
     {
       className,
-      type,
       steps,
+      type = MULTI_STEP_TYPES.PRIMARY,
       stepComponentClassName,
       dividerComponentClassName,
-      fulfilledStepIcon,
-      fulfilledStepIconType,
-      isFulfilledStepDisplayNumber,
-      onClick,
-      textPlacement,
+      fulfilledStepIcon = Check,
```

Comment:
Just make sure you don't change here the default props values by mistake (for example undefined => false or Noop => undefined)

Reply:
Because it can lead to unexpected behaviour